src/main/java/org/elasticsearch/river/rabbitmq/RabbitmqRiver.java

```diff
@@ -398,16 +398,24 @@ public void run() {
 
                         if (ordered) {
                             try {
+                                Boolean isRecoverableFailure = false;
                                 if (bulkRequestBuilder.numberOfActions() > 0) {
                                   BulkResponse response = bulkRequestBuilder.execute().actionGet();
                                   if (response.hasFailures()) {
                                     // TODO write to exception queue?
-                                    logger.warn("failed to execute" + response.buildFailureMessage());
+                                      String failMsg = response.buildFailureMessage()
+                                      // Rejected executions due to queue capacity limit should be retried
+                                      isRecoverableFailure = failMsg.contains("queue capacity");
+                                      logger.warn("failed to execute" + failMsg);
                                   }
                                 }
                                 for (Long deliveryTag : deliveryTags) {
                                     try {
-                                        channel.basicAck(deliveryTag, false);
+                                        if (isRecoverableFailure) {
+                                            channel.basicNack(deliveryTag, false, false);
```

Comment:
don't you want requeue = true?

Reply:
oh snap, @hjz yeah i think @tksfz is right... requeue=true for 3rd param?